database/database.go

```diff
@@ -55,22 +56,24 @@ func (db *Database) refreshDbConfig(ahr *httpUtils.AuthenticatedHttpRequester) {
 }
 
 func (db *Database) Replicate(shard, replica string, ahr *httpUtils.AuthenticatedHttpRequester) error {
-	replicaNode := fmt.Sprintf("couchdb@%s", replica)
+	replicaNode := node.At(replica)
 
```

Comment:
I'd add a check on `config` field of `db`. It's common to panic because it's `nil` if it's not properly set. While it's not mandatory it's a nice defensive check.